client/src/includes/userbar.ts

```diff
@@ -11,6 +11,55 @@ This component implements a roving tab index for keyboard navigation
 Learn more about roving tabIndex: https://w3c.github.io/aria-practices/#kbd_roving_tabindex
 */
 
+export const sortAxeViolations = (violations: AxeResults['violations']) => {
+  interface inaccessibleNode {
+    selectedNode: HTMLElement;
+    selectorName: string;
+  }
```

Comment:
Let's use `PascalCase` for interface names. And I know that this interface is only used internally, but I *think* we generally want interfaces/type definitions to be in the outermost scope of the file? Not so sure on that one, though!

```suggestion
interface InaccessibleNode {
  selectedNode: HTMLElement;
  selectorName: string;
}

export const sortAxeViolations = (violations: AxeResults['violations']) => {
```

P.s. ignore this if you managed to reimplement the algorithm as Thibaud mentioned so that the interface is no longer needed